futures-core/src/never.rs

```diff
@@ -0,0 +1,29 @@
+//! Definition and trait implementations for the `Never` type,
+//! a stand-in for the `!` type until it becomes stable.
+
+use core::fmt;
+
+/// A type with no possible values.
+///
+/// This is used to indicate values which can never be created, such as the
+/// error type of infallible futures.
+///
+/// This type is a stable equivalent to the `!` type from `std`.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub enum Never {}
+
+impl Never {
+    /// Convert the `Never` type into any other type.
+    pub fn into_any<T>(self) -> T {
```

Comment:
Having this public function would block changing to `type Never = !;` in the future, similar to [`std::convert::Infallible`](https://doc.rust-lang.org/nightly/std/convert/enum.Infallible.html#future-compatibility) I think it would make sense to plan to do this once `!` is stabilized.

Reply:
That makes sense. I'll remove this function.